### PR TITLE
e2e: remove deprecated template_file dependency for tf

### PR DESCRIPTION
**Pull request by pkazmierczak**
This also allows running tf for our e2e suite locally on darwin.

### CODE REVIEW

File: e2e/terraform/ecs.tftpl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+security_groups = ["${sg_id}"]
+subnets         = ["${subnet_id}"]
```
Comment (tgross) [inline e2e/terraform/ecs.tftpl:2]: Missing the EOF here

File: e2e/terraform/volumes.tftpl [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+capability {
+  access_mode     = "single-node-writer"
+  attachment_mode = "file-system"
+}
+
+capability {
+  access_mode = "single-node-reader"
+  attachment_mode = "file-system"
+}
```
Comment (tgross) [inline e2e/terraform/volumes.tftpl:15]: Missing EOF

File: e2e/terraform/nomad-acls.tf [hunk 3]
```diff
@@ -12,24 +12,20 @@ locals {
...
 resource "null_resource" "bootstrap_nomad_acls" {
   depends_on = [module.nomad_server]
   triggers = {
-    script = data.template_file.bootstrap_nomad_script.rendered
+    script = templatefile("${local.nomad_env} ./scripts/bootstrap-nomad.sh", {})
   }
 
   provisioner "local-exec" {
-    command = data.template_file.bootstrap_nomad_script.rendered
+    command = templatefile("${local.nomad_env} ./scripts/bootstrap-nomad.sh", {})
```
Comment (tgross) [inline e2e/terraform/nomad-acls.tf:25]: The `templatefile` function takes a path, whereas this is a rendered string that was originally the rendered _output_ of a `template_file` block. I _think_ we can probably do without the `templatefile` command here entirely if we verify that the `local.nomad_env` is properly interpolated.

(Plus the `triggers` block above becomes useless once we're no longer responding to that render event.)

Comment (pkazmierczak) [reply e2e/terraform/nomad-acls.tf]: I made a correction in [d467da7](https://github.com/hashicorp/nomad/pull/19313/commits/d467da720c830ad7fd8c6a307b48699c007c4fe9) which I think makes sense.

File: e2e/terraform/nomad-acls.tf [hunk 4]
```diff
@@ -15,21 +15,14 @@ locals {
...
   triggers = {
-    script = data.template_file.bootstrap_nomad_script.rendered
+    script = "${local.nomad_env} ./scripts/bootstrap-nomad.sh"
   }
 
   provisioner "local-exec" {
-    command = data.template_file.bootstrap_nomad_script.rendered
+    command = aws_instance.server.0.public_ip != "" ? local.nomad_env : "echo 'Nomad server not ready yet, skipping bootstrap'"
   }
 }
```
Comment (tgross) [inline e2e/terraform/nomad-acls.tf:24]: I'm not sure what this is for... if there's no public IP, wouldn't we be unable to run the bootstrapping in the first place and therefore just halt when the bootstrap skip fails?

**Review by tgross — APPROVED** [review]
LGTM

**Pull request merged**